docs/sources/tempo/introduction/solutions-with-traces/traces-diagnose-errors.md

```diff
@@ -0,0 +1,83 @@
+---
+description: "Learn how tracing data can help you understand application insights and performance as well as triaging issues in your services and applications."
+keywords:
+  - Grafana
+  - traces
+  - tracing
+title: Diagnose errors with traces
+menuTitle: Diagnose errors with traces
+weight: 400
+---
+
+# Diagnose errors with traces
+
+Traces allow you to quickly diagnose errors in your application, ensuring that you can perform Root Cause Analysis (RCA) on request failures.
+Trace visualizations help you determine the spans in which errors occur, along with the context behind those errors, leading to a lower mean time to repair (MTTR).
+
+## Meet Handy Site Corp
+
+Handy Site Corp, a fake website company, runs an ecommerce application that includes user authentication, a product catalog, order management, payment processing, and other services.
+
+Handy Site’s operations team receives several alerts for their error SLO for monitored endpoints in their services. Using their Grafana dashboards, they notice that there are several issues. The dashboard provides the percentages of errors:
+
+![Dashboard showing errors in services](/media/docs/tempo/intro/traces-error-SLO.png)
+
+More than 5% of the requests from users are resulting in an error across several endpoints, such as `/beholder`, `/owlbear`, and `/illithid`, which causes a degradation in performance and usability.
+It’s imperative for the operations team at Handy Site to quickly troubleshoot the issue.
+
+## Use TraceQL to query data
+
+Tempo has a traces-first query language, [TraceQL](https://grafana.com/docs/tempo/latest/traceql/), that provides a unique toolset for selecting and searching tracing data. TraceQL can match traces based on span and resource attributes, timing, and duration and provides basic aggregate functions.
+
+Handy Site’s services and applications are instrumented for tracing, so they can use TraceQL as a debugging tool. Using three TraceQL queries, the team identifies and validates the root cause of the issue.
+
+### Find HTTP errors
+
+The top-level service, `mythical-requester`, calls other services and deals with requests and responses from their users.
+Using Grafana Explore, the operations team starts with a simple TraceQL query to find all traces from this top-level service, where an HTTP response being sent back to a user is 400 or above, such as `Forbidden`, `Not found`, and other responses that don't return data the user expected.
+
+```traceql
+{ resource.service.name = "mythical-requester" && span.http.status_code >= 400 } | select(span.http.target)
+```
+
+The `select` statement in this query includes an additionally returned field for each matching trace span, namely the value of the `http.target` attribute, the endpoints for the SaaS service.
```

Comment:
```suggestion
The addition of the `select` statement to this query (after the `|`) ensures that the query response includes not only the set of matched spans, but also the `http.target` attribute (i.e. the endpoints for the SaaS service) for each of those spans.
```

suggested rewrite for clarity, but this may be subjective.

Reply:
Good clarification.